pkg/providers/launchtemplate/launchtemplate.go

```diff
@@ -164,11 +164,7 @@ func (p *DefaultProvider) createAMIOptions(ctx context.Context, nodeClass *v1bet
 		return nil, err
 	}
 	// Get constrained security groups
-	securityGroups, err := p.securityGroupProvider.List(ctx, nodeClass)
-	if err != nil {
-		return nil, err
-	}
-	if len(securityGroups) == 0 {
+	if len(nodeClass.Status.SecurityGroups) == 0 {
 		return nil, fmt.Errorf("no security groups exist given constraints")
```

Comment:
Thoughts on modifying this error so that it explicitly says the status? 
```suggestion
		return nil, fmt.Errorf("no security groups are present in the status")
```